ntoskrnl/mm/ARM3/wslist.cpp

```diff
@@ -0,0 +1,503 @@
+/*
+ * PROJECT:         ReactOS Kernel
+ * LICENSE:         BSD-3-Clause (https://spdx.org/licenses/BSD-3-Clause.html)
+ * FILE:            ntoskrnl/mm/ARM3/wslist.cpp
+ * PURPOSE:         Working set list management
+ * PROGRAMMERS:     Jérôme Gardou
+ */
+
+/* INCLUDES *******************************************************************/
+#include <ntoskrnl.h>
+
+#define NDEBUG
+#include <debug.h>
+
+#define MODULE_INVOLVED_IN_ARM3
+#include "miarm.h"
+
+/* GLOBALS ********************************************************************/
+PMMWSL MmWorkingSetList;
+KEVENT MmWorkingSetManagerEvent;
+
+/* LOCAL FUNCTIONS ************************************************************/
+namespace ntoskrnl
+{
+
+class MMWSL_impl : public MMWSL
+{
+private:
+    MMPTE GetPteTemplate()
+    {
+        return (this == MmSystemCacheWorkingSetList) ? ValidKernelPte : ValidKernelPteLocal;
+    }
+
+    ULONG GetNextPageColor()
+    {
```

Comment:
Is that an april fools joke?

Reply:
Why would it be ?

Reply:
> Why would it be ?

You are developing an **NT kernel**, not HaikuOS or your average game/app/whatever.

Reply:
And ?